book/scheduling.md

```diff
@@ -339,6 +328,27 @@ That's because after the task has been removed from the queue, it
 can't be accessed by another thread, so the lock does not need to be
 held while the task is running.
 
+The `setTimeout` code is now thread-safe, but still has yet another bug: if we
+navigate from one page to another, `setTimeout` callbacks still pending on the
+previous page eshouldn't do anything. That is easily fixed by adding a
+`discarded` field on `JSContext` and setting it when loading a new page:
+
+``` {.python}
+class JSContext:
+    def __init__(self, tab):
+        # ...
+        self.discarded = False
```

Comment:
Wait, don't we also need to check `discarded` in `dispatch_setTimeout`? I think that's the whole point!